src/main/java/com/thealgorithms/strings/ReverseWordsInString.java

```diff
@@ -0,0 +1,56 @@
+package com.thealgorithms.strings;
+
+/**
+ * Given an input string s, reverse the order of the words.
+ * A word is defined as a sequence of non-space characters.
+ * The words in s will be separated by at least one space.
+ * Return a string of the words in reverse order concatenated by a single space.
+ * Note that s may contain leading or trailing spaces or multiple spaces between two words.
+ * The returned string should only have a single space separating the words.Do not include any extra spaces
+ */
+
+public final class ReverseWordsInString {
+
+    private ReverseWordsInString() {
+    }
+
+    public static String reverseWordsInString(String s) {
+
+        s = s.replaceAll("\\s+", " ").trim();
+        StringBuilder res = new StringBuilder();
+        int start = 0;
+        int end = 0;
+
+        while (start < s.length()) {
+
+            Integer[] arr = findNextWord(s, start, end);
+            start = arr[0];
+            end = arr[1];
+
+            String word = s.substring(start, end);
+
+            start = end;
+
+            if (res.length() > 0) {
+                res.insert(0, " ");
+                res.insert(0, word);
+            } else {
+                res.append(word);
+            }
+        }
+        return res.toString();
+    }
+
+    private static Integer[] findNextWord(String s, int start, int end) {
+        while (s.charAt(start) == ' ') {
+            start++;
+        }
+        end = start + 1;
+
+        while (end < s.length() && (s.charAt(end) != ' ')) {
+            end++;
+        }
+
+        return new Integer[] {start, end};
+    }
```

Comment:
Is this simpler?
```suggestion
    public static String reverseWordsInString(final String s) {
        var words = s.trim().split("\\s+");
        Collections.reverse(Arrays.asList(words));
        return String.join(" ", words);
    }
```

Make sure to:
```java
import java.util.Arrays;
import java.util.Collections;
```

Reply:
yes this is simple but generally, in DS problems, we avoid using built-in functions as much as possible and try to solve our own. I didn't understand this repo objective so here we need to make the code simpler only.

Reply:
My understanding is that people here implement things in order to learn Java. Knowing what is available in given language and its standard libraries is part of this knowledge. Sure: when someone would like to write a method reverting an array and simply called `Collections.reverse(Arrays.asList(...));` inside, that would not be the desired approach.

Reply:
Thanks for the response but didn't understand below part should i use below method because you said here we are to learn java existing methods i was approaching the problems like we do in data structures rounds in interviews

**when someone would like to write a method reverting an array and simply called Collections.reverse(Arrays.asList(...)); inside, that would not be the desired approach.**

Reply:
What I was trying to say is the following. In case if someone would like to add here a class reversing an array of integers, I would expect that the implementation would be using some `for` loop (and not some functionality from the available libraries). Because reversing an array of integers seems to me a good exercise in using loops.

In summary: I think it is completely fine to use here a _three liner_ like the one suggested.

Reply:
done